README.md

```diff
@@ -5,23 +5,55 @@ A command line tool to auto-generate configuration file code, for use in Xcode p
 The `configen` tool is used to auto-generate configuration code from a property list. It is intended to
 create the kind of configuration needed for external URLs or API keys used by your app. Currently supports both Swift and Objective-C code generation.
 
+# Installation
+
+To add the `configen` tool to your Xcode project you need to download or clone this repository and copy the product from the `configen.xcodeproj` project to the root directory of your project. 
+
+The easiest way to do this is to right click on the `configen` product and select ‘Show in Finder’. This file can then be copied to your directory.
+
+Once you have copied the tool to your directory you are ready to go! Now you need to create the relevant files and set-up your project accordingly. This is outlined below.
+
 # Usage
 
 Before running the `configen` tool, you need to create a mapping file, in which you define the configuration variables you support. For example:
 
 ```swift
-entryPointURL : NSURL
-searchURL : NSURL
+entryPointURL : URL
+enableFileSharing : Bool
 retryCount : Int
 adUnitPrefix : String
-pushKey : String
 analyticsKey : String
 environment : Environment
 ```
 
-Then you need to create a property list file, in which you provide values for each of the keys defined in your mapping file, above.
+(NB: When configuring `configen` for Objective C projects the `NSURL` object must be used)
```

Comment:
this is a good note. But it's made me think a bit more about how the whole thing works. Why do we need to use `NSURL` for Objective-C in the mapping file. But we don't use the other corresponding Objective-C types such as `BOOL` and `Integer`.

Either we should update the code so that the following is true: "When configuring `configen` for Objective-C code generation, use the corresponding Objective-C types such as `NSURL`, `BOOL` and `Integer`" ...

... or we should update the code so the following is true: "The `configen` mapping file uses a custom set of types which map to Swift types. Therefore for Objective-C code generation, you must still use Swift equivalent types in the mapping file."